From 6d391211c4f2c7ad14c2c6cad947698334dc997c Mon Sep 17 00:00:00 2001
From: Sanyog Kale <skale@habana.ai>
Date: Sun, 25 Sep 2022 21:21:21 +0300
Subject: [PATCH] [SW-0] common: store fuse spi slave offset

Currently, slave die spi fuse offset is not stored in rom global
structure and always read from image manifest. This patch stores
it in global data structure by which reading from image manifest
can be avoided based on validation pass or boot desc init flag.

Change-Id: I5ddb97e2a63495004249064058fbff685d89045b
---

diff --git a/arch/gaudi3/arc/spi_flash.c b/arch/gaudi3/arc/spi_flash.c
index d50c584..86864b7 100644
--- a/arch/gaudi3/arc/spi_flash.c
+++ b/arch/gaudi3/arc/spi_flash.c
@@ -52,21 +52,23 @@
 	if (idx == HL_FW_IMG_NOT_FOUND)
 		return;
 
+	arcboot_rom_global.d1_fuse_spi_offset = desc.spi_offset;
+
 	/* read value from bank0 */
 	offset = (EFUSE_SYSCFG_OFFSET * sizeof(uint32_t));
-	d1_syscfg1 = spi_flash_readl(desc.spi_offset + offset);
+	d1_syscfg1 = spi_flash_readl(arcboot_rom_global.d1_fuse_spi_offset + offset);
 
 	/* read value from bank1 and OR with value from bank0 */
 	offset = ((FUSE_WORDS_PER_BANK + EFUSE_SYSCFG_OFFSET) * sizeof(uint32_t));
-	d1_syscfg1 |= spi_flash_readl(desc.spi_offset + offset);
+	d1_syscfg1 |= spi_flash_readl(arcboot_rom_global.d1_fuse_spi_offset + offset);
 
 	/* read value from bank0 */
 	offset = (EFUSE_SYSCFG2_OFFSET * sizeof(uint32_t));
-	d1_syscfg2 = spi_flash_readl(desc.spi_offset + offset);
+	d1_syscfg2 = spi_flash_readl(arcboot_rom_global.d1_fuse_spi_offset + offset);
 
 	/* read value from bank1 and OR with value from bank0 */
 	offset = ((FUSE_WORDS_PER_BANK + EFUSE_SYSCFG2_OFFSET) * sizeof(uint32_t));
-	d1_syscfg2 |= spi_flash_readl(desc.spi_offset + offset);
+	d1_syscfg2 |= spi_flash_readl(arcboot_rom_global.d1_fuse_spi_offset + offset);
 
 	/* Notify slave config fuse value in slave scratchpad 0/1 register */
 	mirror_writel(d1_syscfg1, (mmSYSCFG1_EFUSE_DATA + HL_CFG_SPACE_DIE1_OFFSET));
diff --git a/include/arcboot.h b/include/arcboot.h
index 90d6adb..6893bd0 100644
--- a/include/arcboot.h
+++ b/include/arcboot.h
@@ -98,6 +98,7 @@
 	uint8_t die_id;
 	uint8_t sku; /* 0: Dual die 1: Single die */
 	uint32_t d0_fuse_spi_offset;
+	uint32_t d1_fuse_spi_offset;
 	bool fuse_spi;
 	struct hl_fw_boot_ptrs_ctx boot_ptr_ctx[HL_FW_BOOT_PTR_MAX_IMG];
 	struct hl_fw_sec_info sec_data;
